containers/datalab/run.sh

```diff
@@ -20,27 +20,68 @@
 # command prompt, which is useful for tinkering within the container before
 # manually starting the server.
 
+HERE=$(dirname $0)
 CONTENT=$HOME
 ENTRYPOINT="/datalab/run.sh"
-if [ "$1" != "" ]; then
-  if [ "$1" != "shell" ]; then
-    CONTENT=$1
-    shift
-  fi
-  if [ "$1" == "shell" ]; then
-    ENTRYPOINT="/bin/bash"
-  fi
-fi
+DEVROOT_DOCKER_OPTION=''
+while [ $# -gt 0 ]; do
+  case "$1" in
+    shell)
+      ENTRYPOINT="/bin/bash"
+      ;;
+    --live)
+      # Use --live to make the datalab container use a live copy of the
+      # development directory so your changes are visible to the container
+      # without rebuilding.
+      DEVROOT=$(cd "$HERE" && realpath ../..)
+      DEVROOT_DOCKER_OPTION="-v $DEVROOT:/devroot"
+      echo "Updating build-dev"
+      rsync -a \
+          --exclude web/nb/config \
+          --exclude web/nb/static \
+          --exclude web/nb/templates \
+          $DEVROOT/build/ $DEVROOT/build-dev
+      if [ ! -h $DEVROOT/build-dev/web/nb/node_modules ]; then
+        echo linking node_modules
+        ln -s /datalab/web/node_modules $DEVROOT/build-dev/web/nb/node_modules
+      fi
+      # The files in the subdirs of build/web/nb are just copied from the
+      # source dirs. In order to allow using updated versions of those files
+      # without having to run the build script, we remove them from build-dev
+      # and link them directly to the source dirs.
+      # You need to restart the notebook server in the container to make it
+      # pick up your changes.
+      for dir in config static templates; do
+        if [ ! -h $DEVROOT/build-dev/web/nb/$dir ]; then
+          echo linking $dir
+          ln -s /devroot/sources/web/datalab/$dir $DEVROOT/build-dev/web/nb/$dir
```

Comment:
No, it should be /devroot. Although this symlink is being created on the workstation, it is intended to be resolved from within the container, where the /devroot directory exists. I added a comment to the code.